### ISSUE TITLE
Fork GEF support for Python2 to dedicated repository

### ISSUE THREAD

**Issue by hugsy** [hugsy/gef#456]
This issue starts [Projects/4](https://github.com/hugsy/gef/projects/4).

  * [ ] Create separate repository
  * [ ] Trim code from Python2/3 compat check
  * [ ] Update documentation

**Comment by hugsy** [comment]
Created [`gef-legacy`](https://github.com/hugsy/gef-legacy)

**Comment by hugsy** [comment]
Nope, I don't intend to keep it, the difference must be clear (for users and for us).
What I do to make it transparent is that I added that tiny blob into my `~/.gdbinit` so you never have to worry about loading the wrong GEF:

```
python
import sys
if sys.version.splitlines()[0].startswith("3."):
    gdb.execute("source ~/code/gef/gef.py")
else:
  gdb.execute("source ~/code/gef-legacy/gef.py")
```

**Comment by Grazfather** [comment]
That makes sense. Maybe we can add that as a blurb wherever we mention no more support for py2.

**Comment by hugsy** [comment]
Good idea, like an explicit tag 👍 

I hope there won't be too much doc to write, it's not my fav part ...

**Comment by philomath213** [comment]
I liked the idea to move the GEF support for python2 to a dedicated repository, and trim the py2/py3 compat check code, and I want to help with that.
I checked the gef-legacy repository, I found that you just copy/past the code from the original GEF repository to gef-legacy repository, and all commits are just gone.
so I made a separate repository (it's also a fork of this original one) and do the following:
- made "py2-legacy" branch where I can do the work
- create a Dockerfile in order to build a Ubuntu image where we should run the tests, this image contains gdb_7.12 with python2, users can use also directly
- remove Python2/3 compat check
- fix some errors in unittests and update them
- fix some errors in heap commands

right now all test are passing except "test_cmd_heap_bins_non_main".
please take a look at the repository here [gef-legacy](https://github.com/philomath213/gef-legacy/)

**Comment by Grazfather** [comment]
@hugsy this is a better way to do it, that way we don't lose the history. Maybe you can clone @philomath213 's `gef-legacy` and then force push it to our `gef-legacy`?
